### PR TITLE
Remove static state in ProviderTypeLoader 

### PR DESCRIPTION
**Pull request by xiazen**
Package static state in ProviderTypeLoader into a separate class LoadedProviderTypeLoader, and inject it into DI. 

Prep work for logging migration

### CODE REVIEW

File: src/Orleans/Providers/ProviderLoader.cs [hunk 1]
```diff
@@ -51,7 +52,7 @@ public void LoadProviders(IDictionary<string, IProviderConfiguration> configs, I
             }
 
             // Load providers
-            ProviderTypeLoader.AddProviderTypeManager(t => typeof(TProvider).IsAssignableFrom(t), RegisterProviderType);
+            ProviderTypeLoader.AddProviderTypeManager(t => typeof(TProvider).IsAssignableFrom(t), RegisterProviderType, serviceProvider.GetRequiredService<LoadedProviderTypeLoaders>());
```
Comment (ReubenBond) [inline src/Orleans/Providers/ProviderLoader.cs]: Is it possible to just inject `LoadedProviderTypeLoaders` into the constructor instead?

Comment (xiazen) [inline src/Orleans/Providers/ProviderLoader.cs]: to ProviderLoader's ctor? or to ProviderTypeLoader's ctor?

Comment (ReubenBond) [inline src/Orleans/Providers/ProviderLoader.cs]: Either way, as long as we're minimising the use of `GetService()` and friends

File: src/OrleansRuntime/Core/BootstrapProviderManager.cs [hunk 2]
```diff
@@ -10,11 +10,11 @@ internal class BootstrapProviderManager : IProviderManager
         private readonly PluginManager<IBootstrapProvider> pluginManager;
         private readonly string configCategoryName;
 
-        internal BootstrapProviderManager()
+        internal BootstrapProviderManager(LoadedProviderTypeLoaders loadedProviderTypeLoaders)
```
Comment (ReubenBond) [inline src/OrleansRuntime/Core/BootstrapProviderManager.cs]: ctor needs to be public

**Comment by xiazen** [comment]
@dotnet-bot test netstandard-win-bvt

**Pull request merged**